tests/Tokenizer/Transformer/BraceTransformerTest.php

```diff
@@ -296,6 +302,169 @@ public static function providePre84ProcessCases(): iterable
+            ],
+        ];
+
+        yield 'property hooks: property with default value (array)' => [
+            <<<'PHP'
+                <?php
+                class PropertyHooks
+                {
+                    public $bar = [1,2,3] { // << this one
+                        set($value) {
+                            $this->bar = $value;
+                        }
+                    } // << this one
+                }
+                PHP,
+            [
+                21 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
+                43 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
+            ],
+        ];
+
+        yield 'property hooks: property with default value (namespaced)' => [
+            <<<'PHP'
+                <?php
+                class PropertyHooks
+                {
+                    public $bar = DateTimeInterface::ISO8601 { // << this one
+                        set($value) {
+                            $this->bar = $value;
+                        }
+                    } // << this one
+                }
+                PHP,
+            [
+                17 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
+                39 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
+            ],
+        ];
+
+        yield 'property hooks: property with setter attributes' => [
+            <<<'PHP'
+                <?php
+                class PropertyHooks
+                {
+                    public string $bar { // << this one
+                        #[A]
+                        #[B]
+                        set(string $value) {
+                            $this->bar = strtolower($value);
+                        }
+                    } // << this one
+                }
+                PHP,
+            [
+                13 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
+                48 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
+            ],
+        ];
+
+        yield 'property hooks: property with short setter' => [
+            <<<'PHP'
+                <?php
+                class PropertyHooks
+                {
+                    public string $bar { // << this one
+                        set {
+                            $this->bar = strtolower($value);
+                        }
+                    } // << this one
+                }
+                PHP,
+            [
+                13 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
+                35 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
+            ],
+        ];
+
+        yield 'property hooks: property with short getter' => [
+            <<<'PHP'
+                <?php
+                class PropertyHooks
+                {
+                    public string $bar { // << this one
+                        get => ucwords(mb_strtolower($this->bar));
+                    } // << this one
+                }
+                PHP,
+            [
+                13 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
+                32 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
+            ],
+        ];
```

Comment:
What do you think about testing "more curly" property hook, like this?

```suggestion
        ];

        yield 'property hooks: some more curly braces within hook' => [
            <<<'PHP'
                <?php
                class PropertyHooks
                {
                    public $callable { // << this one
                        set($value) {
                            if (is_callable($value)) {
                                $this->callable = $value;
                            } else {
                                $this->callable = static function (): void {
                                    $foo = new class implements \Stringable {
                                        public function __toString(): string {
                                            echo 'Na';
                                        }
                                    };

                                    for ($i = 0; $i < 8; $i++) {
                                        echo (string) $foo;
                                    }
                                };
                            }
                        }
                    } // << this one
                }
                PHP,
            [
                11 => CT::T_PROPERTY_HOOK_BRACE_OPEN,
                143 => CT::T_PROPERTY_HOOK_BRACE_CLOSE,
            ],
        ];
```

It covers `naivelyFindCurlyBlockEnd()` more than actual transformer, but since it's a private method of it and we need to make sure we determine end of the block properly, maybe such test case is valuable.